lib/record/absoluteUri.dart

```diff
@@ -15,16 +15,12 @@ class AbsoluteUriRecord extends NDEFRecord {
     }
   }
 
-  String get uri {
-    return decodedType;
+  String? get uri {
+    return decodedType; // This is really bad hack
```

Comment:
I might be wrong but, is this `type` and its friends more an `WellKnownRecord` thing rather than a generic thing in `NDEFRecord`?

The code here seems abuse the `type` to store the `uri`, and caller side still see strange things, such as `record.type` doesn't really return its type but an URI string.

It might be a bigger change, but it is going to the things right

Reply:
The `AbsoluteURI` record uses `Payload Type` as `uri` data. I confirmed it through using an android phone to read a card with Absolute URI, and the phone advises a browser app which can open the `uri` stored in `type`. 

`record.type` is designed to be consistent with NDEF standard. You might use `record.tnf` (`Type Name Format` in NDEF) to distinguish different record types.

You can see this article for further information about NDEF `AbsoluteURI` record: [Chapter 4. Introducing NDEF](https://www.oreilly.com/library/view/beginning-nfc/9781449324094/ch04.html)